components/integrations/abstract_incoming_webhook.tsx

```diff
@@ -10,64 +9,90 @@ import BackstageHeader from 'components/backstage/components/backstage_header';
 import ChannelSelect from 'components/channel_select';
 import FormError from 'components/form_error';
 import SpinnerButton from 'components/spinner_button';
+import {Team} from '@mattermost/types/teams';
 import {localizeMessage} from 'utils/utils';
 
-export default class AbstractIncomingWebhook extends React.PureComponent {
-    static propTypes = {
-
-        /**
-        * The current team
-        */
-        team: PropTypes.object.isRequired,
-
-        /**
-        * The header text to render, has id and defaultMessage
-        */
-        header: PropTypes.object.isRequired,
-
-        /**
-        * The footer text to render, has id and defaultMessage
-        */
-        footer: PropTypes.object.isRequired,
-
-        /**
-        * The spinner loading text to render, has id and defaultMessage
-        */
-        loading: PropTypes.object.isRequired,
-
-        /**
-        * The server error text after a failed action
-        */
-        serverError: PropTypes.string.isRequired,
-
-        /**
-        * The hook used to set the initial state
-        */
-        initialHook: PropTypes.object,
-
-        /**
-        * Whether to allow configuration of the default post username.
-        */
-        enablePostUsernameOverride: PropTypes.bool.isRequired,
-
-        /**
-        * Whether to allow configuration of the default post icon.
-        */
-        enablePostIconOverride: PropTypes.bool.isRequired,
-
-        /**
-        * The async function to run when the action button is pressed
-        */
-        action: PropTypes.func.isRequired,
-    }
+interface Hook {
```

Comment:
Instead of adding a new type for this, could we use `IncomingWebhook` from `@mattermost/types/integrations` for this or `Partial<IncomingWebhook>`?